### PR TITLE
Add Category and Topic API (+ tests)

### LINKED ISSUE #7: Jest Based Unit as well as Integration Testing
Working on it, using JestJS for Testing.

### LINKED ISSUE #14: Category and Topic API
To-do:

Improve Category, Topic Model and Graphql
Add Organization Model
Add Category and Topic APIs - CRUD operations, archiving collections, getting self-created Categories and Topics, fetching messages from a particular topic

### PR DESCRIPTION
**Pull request by aa0501**
Fixes #7 #14 
Made on top of #13 (to be merged after it)

### CODE REVIEW

**Comment by aa0501** [comment]
Added 38 tests for APIs and improved overall file structure. Consider User, Organization, Category and Topic API basics complete after this PR.
Tests status:
image

**Comment by vaibhavdaren** [comment]
"rebase --> e104887" properly from next time

File: example.env [hunk 1]
```diff
@@ -1,3 +1,4 @@
+NODE_ENV = "development"
 DATABASE_URL="mongodb://<username>:<password>@cluster0-shard-00-00.t4hho.mongodb.net:27017,cluster0-shard-00-01.t4hho.mongodb.net:27017,cluster0-shard-00-02.t4hho.mongodb.net:27017/<dbname>?ssl=true&replicaSet=atlas-a80apd-shard-0&authSource=admin&retryWrites=true&w=majority"
 JWT_SECRET = "somesupersecretkey"
 PORT=8000
```
Comment (vaibhavdaren) [inline example.env:4]:
```suggestion
PORT=8000

```

File: example.env.test [hunk 2]
```diff
@@ -0,0 +1,4 @@
+NODE_ENV = "testing"
+DATABASE_URL="mongodb://<username>:<password>@cluster0-shard-00-00.t4hho.mongodb.net:27017,cluster0-shard-00-01.t4hho.mongodb.net:27017,cluster0-shard-00-02.t4hho.mongodb.net:27017/<dbname>?ssl=true&replicaSet=atlas-a80apd-shard-0&authSource=admin&retryWrites=true&w=majority"
+JWT_SECRET = "somesupersecretkey"
+PORT=8080
```
Comment (vaibhavdaren) [inline example.env.test:4]:
```suggestion
PORT=8080

```

**Pull request merged**